packages/react-reconciler/src/ReactChildFiber.new.js

```diff
@@ -1365,13 +1471,59 @@ function createChildReconciler(shouldTrackSideEffects): ChildReconciler {
     return deleteRemainingChildren(returnFiber, currentFirstChild);
   }
 
-  return reconcileChildFibers;
+  return reconcileChildFibersImpl;
 }
 
-export const reconcileChildFibers: ChildReconciler = createChildReconciler(
-  true,
-);
-export const mountChildFibers: ChildReconciler = createChildReconciler(false);
+const reconcileChildFibersImpl: ChildReconciler = createChildReconciler(true);
+const mountChildFibersImpl: ChildReconciler = createChildReconciler(false);
+
+export function reconcileChildFibers(
+  returnFiber: Fiber,
+  currentFirstChild: Fiber | null,
+  newChild: any,
+  lanes: Lanes,
+): Fiber | null {
+  // This indirection only exists so we can reset `thenableState` at the end.
+  // It should get inlined by Closure.
```

Comment:
This doesn't appear to be the case. This function is called in many places and it's larger than closure considers inlinable. reconcileChildFibersImpl is returned as first class so it can't be inlined and is also recursive, so it's also not inlined into this function.

Reply:
Hmm maybe it would work if I move the indirection into the factory function

I could also submit a PR to get rid of `createChildReconciler` and thread the arguments through instead

Reply:
It makes sense to me that `reconcileChildFibersImpl` wouldn't get inlined, since it's recursive. Do you have a guess for why the outer one (`reconcileChildFibers`) isn't inlined into its callers, though? Since it's only a few lines long.